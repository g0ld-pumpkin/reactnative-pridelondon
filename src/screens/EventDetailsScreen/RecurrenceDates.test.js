// @flow
import React from "react";
import { shallow } from "enzyme";
import RecurrenceDates from "./RecurrenceDates";

it("renders correctly", () => {
  const props = {
    recurrenceDates: [
      "10/07/2018",
      "11/07/2018",
      "12/07/2018",
      "13/07/2018",
      "14/07/2018",
      "15/07/2018"
    ],
    startTime: "2018-07-09T11:00+01:00"
  };
  const output = shallow(<RecurrenceDates {...props} />);
  expect(output).toMatchSnapshot();
});

it("renders day range for given recurring dates", () => {
  const props = {
    recurrenceDates: [
      "10/07/2018",
      "11/07/2018",
      "12/07/2018",
      "13/07/2018",
      "14/07/2018",
      "15/07/2018"
    ],
    startTime: "2018-07-09T11:00+01:00"
  };
  const output = shallow(<RecurrenceDates {...props} />);

  expect(output.children().text()).toContain("9 Jul - 15 Jul");
});

it("renders day range for recurring dates that are out of order", () => {
  const props = {
    recurrenceDates: [
      "10/07/2018",
      "13/07/2018",
      "11/07/2018",
      "14/07/2018",
      "12/07/2018",
      "15/07/2018"
    ],
    startTime: "2018-07-09T11:00+01:00"
  };
  const output = shallow(<RecurrenceDates {...props} />);

  expect(output.children().text()).toContain("9 Jul - 15 Jul");
});

it("renders nothing if no dates provided", () => {
  const props = {
    recurrenceDates: [],
    startTime: "2018-07-09T11:00+01:00"
  };
  const output = shallow(<RecurrenceDates {...props} />);

  expect(output.children().length).toBe(0);
});
