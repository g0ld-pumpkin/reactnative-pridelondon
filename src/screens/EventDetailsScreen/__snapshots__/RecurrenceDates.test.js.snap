// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  color="blackColor"
  markdown={false}
  markdownStyle={Object {}}
  style={
    Object {
      "fontStyle": "italic",
    }
  }
  type="small"
>
  Runs from: 9 Jul - 15 Jul
</Text>
`;
